static/app/utils/queryClient.tsx

```diff
@@ -17,6 +18,24 @@ type UseQueryOptions<TQueryFnData, TError, TData, TQueryKey extends QueryKey> =
   'queryKey' | 'queryFn'
 >;
 
+interface UseMutationOptionsWithApiOptions<TData, TError, TVariables, TContext>
+  extends Omit<
+    reactQuery.UseMutationOptions<TData, TError, TVariables, TContext>,
+    'mutationFn'
+  > {
+  api: {
+    method: Exclude<APIRequestMethod, 'GET'>;
+    url: string;
+    data?: any;
```

Comment:
would this type be an object? maybe we can 
```suggestion
    data?: Record<string, any>;
```
?

Reply:
The body of the request can be any json value, so it can be a lot of things. The real correct typing would be something like this `JSONValue` here: https://github.com/sindresorhus/type-fest/blob/main/source/basic.d.ts

But IDK if it matters too much here if we perfectly type it - `api.tsx` also uses `any`